responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -586,14 +586,22 @@ def _validate_rai_insights_input_parameters(
                 "identified as categorical features: "
                 f"{non_categorical_or_time_string_columns}")
 
+        # Check if any of the data is missing in test and train data
         list_of_feature_having_missing_values = []
         for feature in test.columns.tolist():
             if np.any(test[feature].isnull()):
                 list_of_feature_having_missing_values.append(feature)
         if len(list_of_feature_having_missing_values) > 0:
-            warnings.warn(
+            raise UserConfigValidationException(
                 f"Features {list_of_feature_having_missing_values} "
                 "have missing values in test data")
+        for feature in train.columns.tolist():
+            if np.any(train[feature].isnull()):
+                list_of_feature_having_missing_values.append(feature)
+        if len(list_of_feature_having_missing_values) > 0:
+            raise UserConfigValidationException(
```

Comment:
Some ML models, like LightGBM, can actually handle missing values and predict on them - and certainly many pipelines can if they have some imputer as one of the steps (and we can take a whole pipeline as input).  I've had customers in the past complain that we should just be able to handle missing values.  I guess I'm ok with this change but I'm really worried what we will have to tell our customers when we get such issues - often they just want everything to 'just work' and they are reluctant to take a bunch of extra steps just to get something to work in our dashboard when their pipeline already works end to end on their data with missing values.

Reply:
makes sense; tagged you in a comment below!